src/abstractdataframe/abstractdataframe.jl

```diff
@@ -67,22 +67,48 @@ _names(df::AbstractDataFrame) = _names(index(df))
 
 Compat.hasproperty(df::AbstractDataFrame, s::Symbol) = haskey(index(df), s)
 
-function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol}; makeunique::Bool=false)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol};
+                 makeunique::Bool=false)
     rename!(index(df), vals, makeunique=makeunique)
     return df
 end
 
-function rename!(df::AbstractDataFrame, args...)
-    rename!(index(df), args...)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{<:AbstractString};
+                 makeunique::Bool=false)
+    rename!(index(df), Symbol.(vals), makeunique=makeunique)
     return df
 end
+
+rename!(df::AbstractDataFrame, args::AbstractVector{Pair{Symbol,Symbol}}) =
+    rename!(index(df), args)
+
+function rename!(df::AbstractDataFrame,
+                 args::Union{AbstractVector{<:Pair{Symbol,<:AbstractString}},
+                             AbstractVector{<:Pair{<:AbstractString,Symbol}},
+                             AbstractVector{<:Pair{<:AbstractString,<:AbstractString}},
+                             AbstractDict{Symbol,Symbol},
+                             AbstractDict{Symbol,<:AbstractString},
+                             AbstractDict{<:AbstractString,Symbol},
+                             AbstractDict{<:AbstractString,<:AbstractString}})
+    args_vec = Vector{Pair{Symbol, Symbol}}(undef, length(args))
```

Comment:
I realize this could just be `[Symbol(from) => Symbol(to) for (from, to) in args]`, right?

Reply:
sure - I do not know why I did not use this pattern 😄.